packages/store/src/internal/state-factory.ts

```diff
@@ -211,6 +212,28 @@ export class StateFactory {
     return forkJoin(results);
   }
 
+  private checkDuplicateStates(stateClasses: StateClass[]): StateClass[] {
```

Comment:
> but it only checks within the states being added.
Nope, because ngxsBootstrap during the initialization phase, it catches the states that you add to the root or lazy modules